File: core/src/main/scala/ai/h2o/sparkling/backend/api/scalainterpreter/ScalaCodeResult.java
package ai.h2o.sparkling.backend.api.scalainterpreter;

import water.Key;
import water.Lockable;

/**
 * This object is returned by jobs executing the Scala code
 */
public class ScalaCodeResult extends Lockable<ScalaCodeResult> {
    public String code;
    public String scalaStatus;
    public String scalaResponse;
    public String scalaOutput;

    /**
     * Create a Lockable object, if it has a {@link Key}.
     *
     * @param key key
     */
    public ScalaCodeResult(Key<ScalaCodeResult> key) {
        super(key);
    }

    @Override
    public Class<ScalaCodeResultV3> makeSchema() {
        return ScalaCodeResultV3.class;
    }
}

File: core/src/main/scala/ai/h2o/sparkling/backend/api/scalainterpreter/ScalaCodeResult.scala
package ai.h2o.sparkling.backend.api.scalainterpreter

import ai.h2o.sparkling.backend.api.ParameterBase
import javax.servlet.http.HttpServletRequest;

/**
  * This object is returned by jobs executing the Scala code
  */
case class ScalaCodeResult(code: String, scalaStatus: String, scalaResponse: String, scalaOutput: String)

object ScalaCodeResult extends ParameterBase {

  private[scalainterpreter] case class ScalaCodeResultParameters(resultKey: String) {
    def validate(): Unit = {}
  }

  object ScalaCodeResultParameters {
    private[scalainterpreter] def parse(request: HttpServletRequest): ScalaCodeResultParameters = {
      val resultKey = request.getPathInfo.drop(1).split("/")(1)
      ScalaCodeResultParameters(resultKey)
    }
  }
}
